### PR TITLE
check node before prefetching

### PR DESCRIPTION
**Pull request by adiprerepa**
follow-up #298

### CODE REVIEW

**Comment by adiprerepa** [comment]
I really really really really don't like the four nested if statements but Rust won't me chain `if let` and `if <expr>` together.

File: src/workload.rs [hunk 1]
```diff
@@ -559,9 +563,13 @@ impl WorkloadStore {
                     .insert((service_sock_addr, port.target_port as u16));
             }
         }
-        if let Some(tx) = self.cert_tx.as_mut() {
-            if let Err(e) = tx.try_send(widentity) {
-                info!("couldn't prefetch: {:?}", e)
+        if let Some(node) = self.local_node.clone() {
+            if node == w.clone().node {
```
Comment (howardjohn) [inline src/workload.rs:567]: Replace with `if Some(w.node) == self.local_node`. There should be clones needed

Comment (adiprerepa) [reply src/workload.rs]: done

File: src/workload.rs [hunk 2]
```diff
@@ -559,9 +563,11 @@ impl WorkloadStore {
                     .insert((service_sock_addr, port.target_port as u16));
             }
         }
-        if let Some(tx) = self.cert_tx.as_mut() {
-            if let Err(e) = tx.try_send(widentity) {
-                info!("couldn't prefetch: {:?}", e)
+        if Some(w.node) == self.local_node.clone() {
```
Comment (howardjohn) [inline src/workload.rs:566]: why do we need to clone? if its needed can we do `Some(&w.node) == self.local_node.as_ref`

Comment (adiprerepa) [reply src/workload.rs]: if I just did `self.local_node` it would error out with `cannot move`. Changed it to that though.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**